### ISSUE TITLE
When ESBMC output is too big, a TPM Rate Limit error occurs.

### ISSUE THREAD

**Issue by Yiannis128** [Yiannis128/esbmc-ai#93]
Example:

```
Retrying langchain.chat_models.openai.ChatOpenAI.completion_with_retry.<locals>._completion_with_retry in 4.0 seconds as it raised RateLimitError: Rate limit reached for gpt-3.5-turbo-16k in organization org-k00KcqV3VgXTcgFIkNM9NWHJ on tokens per min (TPM): Limit 180000, Used 150144, Requested 53468. Please try again in 7.87s. Visit https://platform.openai.com/account/rate-limits to learn more..
```

**Comment by ishaan-jaff** [comment]
@Yiannis128 
i'm the maintainer of LiteLLM we allow you to increase your throughput -  load balance between multiple deployments (Azure, OpenAI)
**I'd love your feedback, especially if this does not solve your problem** 

Here's how to use it
Docs: https://docs.litellm.ai/docs/routing


```python
from litellm import Router

model_list = [{ # list of model deployments 
    "model_name": "gpt-3.5-turbo", # model alias 
    "litellm_params": { # params for litellm completion/embedding call 
        "model": "azure/chatgpt-v-2", # actual model name
        "api_key": os.getenv("AZURE_API_KEY"),
        "api_version": os.getenv("AZURE_API_VERSION"),
        "api_base": os.getenv("AZURE_API_BASE")
    }
}, {
    "model_name": "gpt-3.5-turbo", 
    "litellm_params": { # params for litellm completion/embedding call 
        "model": "azure/chatgpt-functioncalling", 
        "api_key": os.getenv("AZURE_API_KEY"),
        "api_version": os.getenv("AZURE_API_VERSION"),
        "api_base": os.getenv("AZURE_API_BASE")
    }
}, {
    "model_name": "gpt-3.5-turbo", 
    "litellm_params": { # params for litellm completion/embedding call 
        "model": "vllm/TheBloke/Marcoroni-70B-v1-AWQ", 
        "api_key": os.getenv("OPENAI_API_KEY"),
    }
}]

router = Router(model_list=model_list)

# openai.ChatCompletion.create replacement
response = router.completion(model="gpt-3.5-turbo", 
                messages=[{"role": "user", "content": "Hey, how's it going?"}])

print(response)
```

**Comment by Yiannis128** [comment]
> @Yiannis128 
> i'm the maintainer of LiteLLM we allow you to increase your throughput -  load balance between multiple deployments (Azure, OpenAI)
> **I'd love your feedback, especially if this does not solve your problem** 
> 
> Here's how to use it
> Docs: https://docs.litellm.ai/docs/routing
> 
> 
> ```python
> from litellm import Router
> 
> model_list = [{ # list of model deployments 
>     "model_name": "gpt-3.5-turbo", # model alias 
>     "litellm_params": { # params for litellm completion/embedding call 
>         "model": "azure/chatgpt-v-2", # actual model name
>         "api_key": os.getenv("AZURE_API_KEY"),
>         "api_version": os.getenv("AZURE_API_VERSION"),
>         "api_base": os.getenv("AZURE_API_BASE")
>     }
> }, {
>     "model_name": "gpt-3.5-turbo", 
>     "litellm_params": { # params for litellm completion/embedding call 
>         "model": "azure/chatgpt-functioncalling", 
>         "api_key": os.getenv("AZURE_API_KEY"),
>         "api_version": os.getenv("AZURE_API_VERSION"),
>         "api_base": os.getenv("AZURE_API_BASE")
>     }
> }, {
>     "model_name": "gpt-3.5-turbo", 
>     "litellm_params": { # params for litellm completion/embedding call 
>         "model": "vllm/TheBloke/Marcoroni-70B-v1-AWQ", 
>         "api_key": os.getenv("OPENAI_API_KEY"),
>     }
> }]
> 
> router = Router(model_list=model_list)
> 
> # openai.ChatCompletion.create replacement
> response = router.completion(model="gpt-3.5-turbo", 
>                 messages=[{"role": "user", "content": "Hey, how's it going?"}])
> 
> print(response)
> ```

Hi, thanks for the suggestion, before I look at this, I would like to ask if you have a hugging face model uploaded? Since I already have hugging face model support.

I will still look at it if you don't but if you do it will be much easier to implement.

**Comment by ishaan-jaff** [comment]
Yes we support hugging face llms -  are you trying to load balance between hugging face endpoints ?

**Comment by Yiannis128** [comment]
> Yes we support hugging face llms - are you trying to load balance between hugging face endpoints ?

No, I only ask it because I have an interface for adding text-generation-inference compatible models through hugging face. So if you do, which is great! Is this an alternative to langchain? Could you inform me of the positives you have?
